server/index.js

```diff
@@ -40,16 +38,105 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.status(200).json([{}]);
+  } else {
+    const value = [req.session.cartId];
+    const sql = `
+        select "c"."cartItemId",
+                "c"."price",
+                "p"."productId",
+                "p"."image",
+                "p"."name",
+                "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+        where "c"."cartId" = $1
+    `;
+    db.query(sql, value)
+      .then(result => {
+        JSON.stringify(result);
+        res.status(202);
+        res.json(result.rows[0]);
+      })
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  if (parseInt(req.body.productId)) {
+    const myVal = [req.body.productId];
+    const sql = `
+      select "price"
+      from "products"
+      where "productId" = $1
+    `;
+    db.query(sql, myVal)
+      .then(result => {
+        if (req.session.cartId) {
+          const obj = {
+            cartId: req.session.cartId,
+            price: result.rows[0].price
+          };
+          return (obj);
+        } else if (!req.session.cartId && result.rows.length !== 0) {
+          const sql2 = `
+            insert into "carts" ("cartId", "createdAt")
+            values (default, default)
+            returning "cartId"
+          `;
+          return db.query(sql2)
+            .then(ress => {
+              const obj = {
+                cartId: ress.rows[0].cartId,
+                price: result.rows[0].price
+              };
+              return (obj);
+            });
+        } else {
+          next(new ClientError('couldnt get a price for that productId...', 400));
```

Comment:
Be sure to check if the product was found first, then `throw` this error if not. (with `throw`, not `next()`)